src/main/java/sleepy/tasks/Deadline.java

```diff
@@ -0,0 +1,54 @@
+package sleepy.tasks;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * This class is a type of task.
+ *
+ * @author kjw142857
+ */
+public class Deadline extends Task {
+    private String givenDeadline;
+    private LocalDate formattedDate;
+
+    /**
+     * Constructor for the Deadline class.
+     *
+     * @param description Description of the deadline details.
+     * @param givenDeadline When the given deadline is.
+     */
+    public Deadline(String description, String givenDeadline) {
+        super(description);
+        this.givenDeadline = givenDeadline;
+        try {
+            formattedDate = LocalDate.parse(givenDeadline);
+        } catch (DateTimeParseException d) {
+            formattedDate = null;
+        }
+    }
```

Comment:
LGTM. I like the way you construct your constructor and parse the String to LocalDate inside the constructor.